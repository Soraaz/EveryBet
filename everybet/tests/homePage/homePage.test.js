import React from "react";
import renderer from "react-test-renderer";

import HomePage from "../../src/homePage/homePage";

describe("HomePage", () => {
	describe("rendering", () => {
		it("Should render correctly", () => {
			const tree = renderer
				.create(<HomePage />)
				.toJSON();
			expect(tree).to.matchSnapshot();
		});
	});

	describe("functionalities", () => {

		let component;

		beforeEach(() => {
			component = new HomePage();
		});

		it("Should instantiate the class", () => {
			expect(component).to.be.an.instanceOf(HomePage);
		});
	});
});
