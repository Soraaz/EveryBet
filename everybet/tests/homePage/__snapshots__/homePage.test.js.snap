// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage rendering Should render correctly 1`] = `
<RCTSafeAreaView
  style={
    Array [
      Object {
        "flex": 1,
        "justifyContent": "flex-start",
      },
      Object {
        "backgroundColor": "#37597F",
      },
    ]
  }
>
  <Modal
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#00000040",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#ffffff",
              "borderRadius": 5,
              "width": "70%",
            },
            Object {
              "paddingBottom": 10,
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "fontSize": 15,
              "fontWeight": "bold",
              "marginTop": 10,
              "textAlign": "center",
            }
          }
        >
          Récompense quotidienne
        </Text>
        <View
          style={
            Object {
              "borderBottomColor": "#000000",
              "borderBottomWidth": 1,
              "marginVertical": 10,
              "width": "50%",
            }
          }
        />
        <Text
          style={
            Object {
              "fontSize": 12,
              "fontWeight": "bold",
              "marginHorizontal": 10,
              "textAlign": "center",
            }
          }
        >
          Vous vous êtes connecté à EveryBet undefined jours d'affilé !
        </Text>
        <Text
          style={
            Object {
              "fontSize": 12,
              "fontWeight": "bold",
              "marginHorizontal": 10,
              "textAlign": "center",
            }
          }
        >
          Vous avez gagné undefined
          <Image
            source={
              Object {
                "testUri": "../../../assets/images/betcoin_icon.png",
              }
            }
            style={
              Object {
                "height": 12,
                "width": 12,
              }
            }
          />
           comme cadeau pour votre fidélité.
        </Text>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "center",
              "backgroundColor": "#FFFFFF",
              "borderRadius": 10,
              "height": 45,
              "justifyContent": "center",
              "marginTop": 20,
              "opacity": 1,
              "shadowColor": "#000000",
              "shadowOffset": Object {
                "height": 6,
                "width": 0,
              },
              "shadowOpacity": 0.33,
              "shadowRadius": 7.5,
              "width": 215,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 15,
                "textAlign": "center",
              }
            }
          >
            Merci !
          </Text>
        </View>
      </View>
    </View>
  </Modal>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#37597F",
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "flex-start",
      }
    }
  >
    <Image
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      source={
        Object {
          "testUri": "../../../assets/images/everybet_logo.png",
        }
      }
      style={
        Object {
          "height": 39.25,
          "marginLeft": 25,
          "width": 138,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "borderRadius": 10,
          "flexDirection": "row",
          "height": 20,
          "justifyContent": "flex-start",
          "marginLeft": "auto",
          "width": 90,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../assets/images/betcoin_icon.png",
          }
        }
        style={
          Object {
            "height": 30,
            "width": 30,
          }
        }
      />
      <Text
        style={
          Object {
            "fontFamily": "System",
            "fontSize": 10,
            "marginLeft": "auto",
            "marginRight": 5,
            "textAlign": "right",
          }
        }
      >
        0
      </Text>
    </View>
    <Image
      accessible={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      source={
        Object {
          "testUri": "../../../assets/images/user_placeholder.png",
        }
      }
      style={
        Object {
          "borderRadius": 20,
          "height": 40,
          "marginHorizontal": 15,
          "width": 40,
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "justifyContent": "flex-start",
        },
        Object {
          "backgroundColor": "#F4F4FE",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#37597F",
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "flex-start",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "marginHorizontal": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
            }
          }
        >
          Catégorie
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 5,
              "height": 25,
              "justifyContent": "center",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
                "paddingLeft": 10,
              }
            }
          >
            <Text>
              Toutes
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
            "justifyContent": "flex-start",
            "marginHorizontal": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
            }
          }
        >
          Ordre
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderRadius": 5,
              "height": 25,
              "justifyContent": "center",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
                "paddingLeft": 10,
              }
            }
          >
            <Text>
              Plus récents
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
